sdk/core/azure-core/src/http/curl/curl_connection_private.hpp

```diff
@@ -128,6 +128,11 @@ namespace Azure { namespace Core { namespace Http {
       // Get the socket that libcurl is using from handle. Will use this to wait while
       // reading/writing
       // into wire
+#if !defined(_MSC_VER) && (LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR < 45 || LIBCURL_VERSION_MAJOR < 7)
```

Comment:
nit: to be explicit about boolean expression ordering, add brackets.

```suggestion
#if !defined(_MSC_VER) && ((LIBCURL_VERSION_MAJOR == 7 && LIBCURL_VERSION_MINOR < 45) || LIBCURL_VERSION_MAJOR < 7)
```

Also nit: consider adding a comment in source explaining why we have this if-def, and why for this particular libcurl version.

Reply:
And while you are updating this file, run clang-format:

```text
Some files were not formatted correctly according to the .clang-format file.
Please run clang-format version 10 or greater to fix the issue by using this bash command at the root of the repo:
find ./sdk \( -iname '*.hpp' -o -iname '*.cpp' \) ! -iname 'json.hpp' -exec clang-format -i {} \;

List of files not formatted correctly:
sdk/core/azure-core/src/http/curl/curl_connection_private.hpp
```